subprojects/plugins/src/main/java/org/gradle/api/internal/java/JavaLibrary.java

```diff
@@ -36,6 +37,9 @@
  * A SoftwareComponent representing a library that runs on a java virtual machine.
  */
 public class JavaLibrary implements SoftwareComponentInternal {
+    private final static Usage API = DefaultObjectFactory.INSTANCE.named(Usage.class, Usage.JAVA_API);
```

Comment:
❌ ObjectFactory takes care of caching, so it would be more idiomatic to inject it in the constructor and call `named()` there instead of using `DefaultObjectFactory.Instance`. The latter is really a clunky workaround that we want to avoid as much as possible.